app/Jobs/FetchCollectionVolume.php

```diff
@@ -44,9 +45,26 @@ public function handle(): void
             contractAddress: $this->collection->address
         );
 
-        $this->collection->update([
-            'volume' => $volume,
-        ]);
+        DB::transaction(function () use ($volume) {
+            if ($volume !== null) {
+                $this->collection->volumes()->create([
+                    'volume' => $volume,
+                ]);
+
+                $this->collection->update([
+                    'volume' => $volume,
+                    'avg_volume_24h' => $this->collection->averageVolumeSince(now()->subDays(1)),
+                    'avg_volume_7d' => $this->collection->averageVolumeSince(now()->subDays(7)),
+                    'avg_volume_1m' => $this->collection->averageVolumeSince(now()->subMonths(1)),
```

Comment:
issue with this is not having data for the first month but only slowly adding to it while the UI shows 1d/7d/30d options

Reply:
I think we may need to rely on the approach in #548 at first where we update data for 1/7/30d based on the API responses and after a month we can calculate it ourselves based on the volume that is stored to reduce the amount of requests required

Reply:
updated the code to handle situations where we don't have enough data (newly added collections)